pkg/nuctl/command/export.go

```diff
@@ -0,0 +1,306 @@
+	}
+
+	cmd := &cobra.Command{
+		Use:     "project name",
+		Aliases: []string{"proj"},
+		Short:   "Export project with all it's functions and function events",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// if we got positional arguments
+			if len(args) != 0 {
+
+				// second argument is a resource name
+				commandeer.getProjectsOptions.Meta.Name = args[0]
+			}
+
+			// initialize root
+			if err := exportCommandeer.rootCommandeer.initialize(); err != nil {
+				return errors.Wrap(err, "Failed to initialize root")
+			}
+
+			// get namespace
+			commandeer.getProjectsOptions.Meta.Namespace = exportCommandeer.rootCommandeer.namespace
+
+			projects, err := exportCommandeer.rootCommandeer.platform.GetProjects(&commandeer.getProjectsOptions)
+			if err != nil {
+				return errors.Wrap(err, "Failed to get projects")
+			}
+
+			if len(projects) == 0 {
+				cmd.OutOrStdout().Write([]byte("No projects found")) // nolint: errcheck
+				return nil
+			}
+
+			// render the projects
+			return commandeer.renderProjects(projects, commandeer.output, cmd.OutOrStdout())
+		},
+	}
+
+	cmd.PersistentFlags().StringVarP(&commandeer.output, "output", "o", OutputFormatJSON, "Output format - \"yaml\", or \"json\"")
+
+	commandeer.cmd = cmd
+
+	return commandeer
+}
+
+func (e *exportProjectCommandeer) renderProjects(projects []platform.Project, format string, writer io.Writer) error {
+
+	rendererInstance := renderer.NewRenderer(writer)
+
+	switch format {
+	case OutputFormatYAML:
+		return e.renderProjectConfig(projects, rendererInstance.RenderYAML)
+	case OutputFormatJSON:
+		return e.renderProjectConfig(projects, rendererInstance.RenderJSON)
+	}
+
+	return nil
+}
+
+func (e *exportProjectCommandeer) getFunctionEvents(functionConfig *functionconfig.Config) ([]platform.FunctionEvent, error) {
+	getFunctionEventOptions := platform.GetFunctionEventsOptions{
+		Meta: platform.FunctionEventMeta{
+			Name:      "",
+			Namespace: functionConfig.Meta.Namespace,
+			Labels: map[string]string{
+				"nuclio.io/function-name": functionConfig.Meta.Name,
+			},
+		},
+	}
+
+	functionEvents, err := e.rootCommandeer.platform.GetFunctionEvents(&getFunctionEventOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	return functionEvents, nil
+}
+
+func (e *exportProjectCommandeer) exportProjectFunctionsAndFunctionEvents(projectConfig *platform.ProjectConfig) (
+	map[string]*functionconfig.Config, map[string]*platform.FunctionEventConfig, error) {
+	getFunctionOptions := &platform.GetFunctionsOptions{
+		Namespace: projectConfig.Meta.Namespace,
+		Labels:    fmt.Sprintf("nuclio.io/project-name=%s", projectConfig.Meta.Name),
+	}
+	functions, err := e.rootCommandeer.platform.GetFunctions(getFunctionOptions)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "Failed to get functions")
+	}
+	functionMap := map[string]*functionconfig.Config{}
+	functionEventMap := map[string]*platform.FunctionEventConfig{}
+
+	for _, function := range functions {
```

Comment:
could use `function.Initialize` from previous comment here

Reply:
Don't understand what you mean here...

Reply:
we discussed offline to make the initialization less code-dup, anyway - irrelevant